extensions/resource-deployment/src/services/resourceTypeService.ts

```diff
@@ -256,17 +236,39 @@ export class ResourceTypeService implements IResourceTypeService {
 	 * Get the ok button text based on the selected options
 	 */
 	private getOkButtonText(resourceType: ResourceType, selectedOptions: { option: string, value: string }[]): string | undefined {
-		if (resourceType.okButtonText && selectedOptions.length === 1) {
-			const optionGiven = `${selectedOptions[0].option}=${selectedOptions[0].value}`;
+		if (resourceType.okButtonText) {
 			for (const possibleOption of resourceType.okButtonText) {
-				if (possibleOption.when === optionGiven || possibleOption.when === undefined || possibleOption.when.toString().toLowerCase() === 'true') {
+				if (this.processWhenClause(possibleOption.when, selectedOptions)) {
 					return possibleOption.value;
 				}
 			}
 		}
 		return loc.select;
 	}
 
+	private processWhenClause(when: string, selectedOptions: { option: string, value: string }[]): boolean {
```

Comment:
Please add doc comments explaining how this processing works in English (most notably what syntax/values it supports since we don't support the same syntax as the rest of the vs code when clauses)